### PR TITLE
Adding stratification as default for classification

### PR DESCRIPTION
**Pull request by vgarciasc**
Stratification is extremely common when separating into train-test or train-validation. This commit modifies the code so stratification is done by default.

### CODE REVIEW

**Comment by eugenioLR** [comment]
I think this isn't quite right, i think specifying the groups doesn't stratify the data. The KFold class should be replaced with StratifiedKFold instead.

I also don't like very much that you are checking the name of the class to stratify or not. I feel there should be a better way to do it. 

Let me take a look and see what i can come up with.

**Comment by eugenioLR** [comment]
We can also take this opportunity to let the user specify how to do the KFold splits, since there apparently are quite a lot of methods implemented in sklearn [(the methods i'm talking about)](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.model_selection).

**Comment by vgarciasc** [comment]
You're correct, the KFold's third parameter is being used as a stratification vector, however it does not have that function. A StratifiedKFold should be used.

As to referencing the "HundredHammersClassifier" as a string, I agree it could be made better. We could simply have the cross-validation function as a parameter of HundredHammersBase, and let it be `KFold` as default for Base, and `StratifiedKFold` as default for Classifier. Then we could make a `_train_test_split(self, X, y)` function that handles all of this logic. This would also take care of allowing the user to specify the folding they want to use, I think.

**Comment by vgarciasc** [comment]
Refactored the code. Now users can pass the cross-validator to be used as a parameter to HH's classifier/regressor classes. As default, classifier uses StratifiedKFold, and regressor uses simple KFold. Examples are included in the new tests added to `test_classification.py` and `test_regression.py`. Please take a look and feel free to comment!

File: hundred_hammers/regressor.py
Comment (eugenioLR) [inline hundred_hammers/regressor.py:1]: The implementation of _stratify_array is the same as the base class, i would remove it

**Review by eugenioLR — CHANGES_REQUESTED** [review]
It seems good to me. I'm going to change a tiny thing and push it.

**Review by eugenioLR — APPROVED** [review]

**Pull request merged**